### DISCUSSION TITLE
@mactaggart Question on mysql changes

### DISCUSSION THREAD

**Discussion by james-deee** [Netflix/conductor#2298] [Q&A]
@mactaggart 
Hi, sorry, I didn't know how else to reach out to you. We are upgrading our Conductor server with our mysql database and it has 10 million workflows. Unfortunately, your schema changes here: https://github.com/Netflix/conductor/commit/ec521688f6177bac1116f7200597b7af145a5920 effectively lock our db for a long time.

Could you give us some insight into these changes? What the positive benefit is here? THanks.

**Comment by mactaggart** [comment]
Hi @james-deee This change was part of a PR that removed the id column from conductor tables. More details and rationale  here: #2013

**Reply by james-deee** [reply]
Thanks @mactaggart . Unfortunately, on very large size DBs, this is extremely slow. We're trying to figure out how we can do this without taking downtime since it's a non-backwards compatible/ breaking change.

**Reply by mactaggart** [reply]
In this case I’m not sure how we can avoid slowness to run the db migrations unless there are improvements we can make to the migration SQL. Is there a particular statement that’s taking longer than expected?
